ztest: ztress: Disable for SMP and tweaks in the test

nordic-krch opened the pull request:
Currently, ztress is not supporting SMP, mainly because of cpu load calculation algorithm. It may be re-enabled in the future.

Additionally, fixed some random failures of the test. Found #41710. Disabled test for `qemu_cortex_a9`.

nordic-krch committed:
testsuite: ztress: Disable ztress for SMP

Currently, ztress is not supporting SMP, mainly because of
cpu load calculation algorithm. It may be re-enabled in the
future.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
tests: ztest: ztress: Disable for qemu_cortex_a9

Disable test for platform since it fails there for unknown
reason.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

evgeniy-paltsev commented on tests/ztest/ztress/prj.conf:
```diff
@@ -1,2 +1,4 @@
 CONFIG_ZTEST=y
 CONFIG_ZTRESS=y
+CONFIG_SMP=n
```
Hmm,

as I remember from previous discussions it's not correct to disable SMP from the test case.
We should either set `CONFIG_MP_NUM_CPUS=1` (so it would be SMP system with only one cpu core) or use special macroses like `ztest_1cpu_unit_test` to run test case on 1 cpu.

nordic-krch replied:
done

nordic-krch committed:
tests: ztest: ztress: Test tweaks and disabling SMP

Disable SMP in the test. Minor tweaks in the test.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
testsuite: ztress: Minor fixes

Fix initial thread priority and move starting of the
test timer after threads initialization to ensure that
it does not expire before threads are initiated.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

nashif commented:
merging as a hotfix for issue in main branch.
